lib/internal/http2/core.js

```diff
@@ -1684,6 +1685,16 @@ class ClientHttp2Session extends Http2Session {
     if (options.waitForTrailers)
       stream[kState].flags |= STREAM_FLAGS_HAS_TRAILERS;
 
+    const signal = options.signal;
+    if (signal) {
+      validateAbortSignal(signal, 'options.signal');
+      const listener = () => stream.destroy();
```

Comment:
I would prefer this to be called with an error.

Reply:
I've been following the discussion on #36048 on the same topic. I don't know enough about the specs and surrounding discussion to have an opinion either way, but over there it looks like it's not being called with an error either.

Whatever the decision is it should probably be consistent in both http and http2 :smile:.

cc @benjamingr

Reply:
This already terminates with an error though and there is a test for that? Destroying the stream early (even without an explicit error on the http2 request) emits an error (right?)

Reply:
Yes, but I think destroying with an explicit aborted `DOMException` like we do elsewhere is good for consistency here.

Reply:
@jasnell but wouldn't the request get aborted with an ECONNRESET (like in the test) anyway?

Reply:
ECONNREFUSED if it's aborted on the same tick, ECONNRESET otherwise. (Also, think that should be enforced by the test?)

Reply:
Test appreciated, I think the errors a refine no?